extension-ui/background.js

```diff
@@ -193,3 +193,198 @@ function showNotification(site, num, options = {}) {
   
   chrome.notifications.create(notificationOptions);
 }
+
+let timeoutId = null;
+let notificationTimeoutId = null;
+let remainingTime = initialTime;
+let lastVisitTime = Date.now();
+let lastVisitDate = new Date().toDateString();
+let isSiteBlocked = false;
+let notificationShown = false;
+let currentTabId = null;
+let currentTabUrl = null;
+let isTabActive = false; 
```

Comment:
why do you need to save data about current tab? you can get it from chrome
for updating browsing time you have to use chrome handlers of tab getting activated and losing focus